analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyToAutoProperty.cs

```diff
@@ -35,56 +35,52 @@ namespace SonarAnalyzer.Rules.CSharp
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var propertyDeclaration = (PropertyDeclarationSyntax)c.Node;
                     var propertySymbol = c.SemanticModel.GetDeclaredSymbol(propertyDeclaration);
-                    if (propertyDeclaration.AccessorList == null ||
-                        propertyDeclaration.AccessorList.Accessors.Count != 2 ||
-                        propertySymbol == null ||
-                        HasDifferentModifiers(propertyDeclaration.AccessorList.Accessors) ||
-                        HasAttributes(propertyDeclaration.AccessorList.Accessors))
+                    if (propertyDeclaration.AccessorList == null
+                        || propertyDeclaration.AccessorList.Accessors.Count != ReadWriteAccessorCount
+                        || propertySymbol == null
+                        || HasDifferentModifiers(propertyDeclaration.AccessorList.Accessors)
+                        || HasAttributes(propertyDeclaration.AccessorList.Accessors))
                     {
                         return;
                     }
 
                     var accessors = propertyDeclaration.AccessorList.Accessors;
                     var getter = accessors.FirstOrDefault(a => a.IsKind(SyntaxKind.GetAccessorDeclaration));
-                    var setter = accessors.FirstOrDefault(a => a.IsKind(SyntaxKind.SetAccessorDeclaration));
+                    var setter = accessors.FirstOrDefault(a => a.IsAnyKind(SyntaxKind.SetAccessorDeclaration, SyntaxKindEx.InitAccessorDeclaration));
 
                     if (getter == null || setter == null)
                     {
                         return;
                     }
-                    if (TryGetFieldFromGetter(getter, c.SemanticModel, out var getterField) &&
-                        TryGetFieldFromSetter(setter, c.SemanticModel, out var setterField) &&
-                        getterField.Equals(setterField) &&
-                        !getterField.GetAttributes().Any() &&
-                        !getterField.IsVolatile &&
-                        getterField.IsStatic == propertySymbol.IsStatic &&
-                        getterField.Type.Equals(propertySymbol.Type))
+                    if (TryGetFieldFromGetter(getter, c.SemanticModel, out var getterField)
```

Comment:
The previous `if return` could be reverted and embedded in to this long condition chain.